File: src/mvp/Algorithm/Dummy/DerivedDummy.cc
#include <mvp/Algorithm/Dummy.h>

namespace mvp {
namespace algorithm {

struct DerivedDummy : public Dummy {
  int m_x;
  int m_y;

  DerivedDummy(int x, int y) : m_x(x), m_y(y) {}

  void void0() {}

  void void1(int a) {}

  void void2(int a, int b) {}

  int function0() { return 0; }

  int function1(int a) { return a; }

  int function2(int a, int b) { return a + b; }

  int x() const { return m_x; }

  int y() const { return m_y; }

  vw::Vector2 do_vector(vw::Vector3 const& a) { return vw::math::subvector(a, 0, 2); }

  vw::Vector2 do_vector(vw::Vector2 const& a) { return a; }
};

REGISTER_ALGORITHM_OBJECT(DerivedDummy);

}} // namespace algorithm,mvp

File: src/mvp/Algorithm/Dummy/DerivedDummy.m
function self = DerivedDummy(x, y)
  self = mvpclass();

  self.m_x = 0;
  self.m_y = 0;

  if (nargin > 0)
    self.m_x = x;
  endif

  if (nargin > 1)
    self.m_y = y;
  endif

  self.void0 = @void0;
  self.void1 = @void1;
  self.void2 = @void2;
  self.function0 = @function0;
  self.function1 = @function1;
  self.function2 = @function2;
  self.x = @x;
  self.y = @y;
endfunction

function void0(self)
  disp("ASDFASDF");
endfunction

function void1(self)
endfunction

function void2(self)
endfunction

function rtn = function0(self)
  rtn = 0;
endfunction

function rtn = function1(self, a)
  rtn = a;
endfunction

function rtn = function2(self, a, b)
  rtn = a + b;
endfunction

function rtn = x(self)
  rtn = self.m_x;
endfunction

function rtn = y(self)
  rtn = self.m_y;
endfunction

% vim:set syntax=octave:
